lesson3/hometask.rb

```diff
@@ -0,0 +1,25 @@
+module MyEach
+  def my_each
+    i = 0
+    while i < self.length
+    	case 
```

Comment:
you can avoid using explicit check of equivalence:
```ruby
      case self
      when Array
        yield self[i]
      when Hash
        yield [self.keys[i], self.values[i]]
      end
      i += 1
```
but it's not the best way to check type of object in module. Ideally, it should be isolated from it.